src/common/components/repositories-home/index.js

```diff
@@ -59,11 +49,16 @@ class RepositoriesHome extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    const snapsLength = nextProps.snaps.ids.length;
-
-    if ((this.props.snaps.success !== nextProps.snaps.success)
```

Comment:
Yup, especially that we don't handle it consistently - sometimes `success` is cleared with every new fetch, sometimes not...